### DISCUSSION TITLE
Nurse scheduling based on calendars

### DISCUSSION THREAD

**Discussion by techhead2** [google/or-tools#4157] [CP-SAT questions]
I had a look at the original nurse scheduling program - https://github.com/google/or-tools/blob/main/examples/python/shift_scheduling_sat.py. I want to modify it so, that shifts are not "allocated" in any random order. In my case E.g a certain operation happens only on Mondays.  The shift which would assist this operation MUST be scheduled for Monday. I really don't know how to start. How can I bring this new dimension into the logic of the program? Any hint/help is appreciated.

**Comment by lperron** [comment]
same answer as on stack overflow. Please provide details, some code...

**Reply by techhead2** [reply]
Can not write code without knowing what to add. I can see in the code that it assigns shifts to employees, but somehow I need to first assign a given shift to a date and then to someone with a given expertee.

**Comment by lperron** [comment]
This is not the way to ask a question.

1) look at the complete example:
https://github.com/google/or-tools/blob/stable/examples/python/shift_scheduling_sat.py
2) in this examples, days are numbered from 0.
3) the shift scheduling example includes both shift assignments and
requests.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00

**Reply by techhead2** [reply]
Yes, but both the assignents and the requests are for or by empleyees. In my case I don't know which employee will do it, just the date when it will happen. This causes me the headache.

**Reply by techhead2** [reply]
As the scheduler program is your brainchild I thought that you would have an offhand answer to this question. Of course I could only write a very stupid piece of code, but that would have nothing positive to my question.

**Comment by lperron** [comment]
This is the request part of the model.
